samples/boards/nrf52/mesh/onoff_level_lighting_vnd_app/src/mesh/device_composition.c

```diff
@@ -1140,48 +1035,67 @@ static bool light_ctl_setunack(struct bt_mesh_model *model,
 			       struct bt_mesh_msg_ctx *ctx,
 			       struct net_buf_simple *buf)
 {
-	u8_t tid;
-	u16_t lightness, temp, delta_uv;
+	u8_t tid, buf_len, delay;
+	s16_t delta_uv;
+	u16_t lightness, temp;
+	s64_t now;
 	struct net_buf_simple *msg = model->pub->msg;
 	struct light_ctl_state *state = model->user_data;
 
+	buf_len = buf->len;
 	lightness = net_buf_simple_pull_le16(buf);
 	temp = net_buf_simple_pull_le16(buf);
-	delta_uv = (int16_t) net_buf_simple_pull_le16(buf);
+	delta_uv = (s16_t) net_buf_simple_pull_le16(buf);
 	tid = net_buf_simple_pull_u8(buf);
 
-	if (state->last_tid == tid && state->last_tx_addr == ctx->addr) {
-		if (temp < TEMP_MIN || temp > TEMP_MAX) {
-			return false;
-		}
+	if (temp < TEMP_MIN || temp > TEMP_MAX) {
+		return false;
+	}
+
+	now = k_uptime_get();
+	if (state->last_tid == tid && state->last_tx_addr == ctx->addr &&
+	    (now - state->last_msg_timestamp <= 6000)) {
 		return true;
 	}
 
-	if (temp < TEMP_MIN || temp > TEMP_MAX) {
+	switch (buf_len) {
```

Comment:
I realize this buf_len stuff is probably coming from the SIGs Mesh guide, but you could avoid the extra variable by looking at buf->len instead, which will tell you how many bytes are left in the buffer (0 or 2 in your case). Same for the other places where you've used buf_len.

Reply:
done !!